File: src/test/java/com/everyonewaiter/domain/auth/JwtPayloadTest.java
package com.everyonewaiter.domain.auth;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;

import org.junit.jupiter.api.Test;

class JwtPayloadTest {

  @Test
  void constructor() {
    JwtPayload payload1 = new JwtPayload(1L, "1");
    JwtPayload payload2 = new JwtPayload(2L, 2L);

    assertThat(payload1.id()).isEqualTo(1L);
    assertThat(payload1.subject()).isEqualTo("1");
    assertThat(payload2.id()).isEqualTo(2L);
    assertThat(payload2.subject()).isEqualTo("2");
  }

  @Test
  void parseLongSubject() {
    JwtPayload payload = new JwtPayload(1L, "1");

    Long subject = payload.parseLongSubject();

    assertThat(subject).isEqualTo(1L);
  }

  @Test
  void parseLongSubjectFail() {
    JwtPayload payload = new JwtPayload(1L, "string");

    assertThatThrownBy(payload::parseLongSubject)
        .isInstanceOf(NumberFormatException.class);
  }

}

File: src/test/java/com/everyonewaiter/domain/auth/JwtPayloadTest.kt
package com.everyonewaiter.domain.auth

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class JwtPayloadTest {

  @Test
  fun `JWT 페이로드의 ID가 숫자인 경우 파싱`() {
    val payload1 = JwtPayload("1", "subject")
    val payload2 = JwtPayload(2L, "subject")

    assertThat(payload1.longId).isEqualTo(1)
    assertThat(payload2.longId).isEqualTo(2)
  }

  @Test
  fun `JWT 페이로드의 ID가 숫자가 아닌 경우 파싱 실패`() {
    val payload = JwtPayload("id", "subject")

    assertThatThrownBy { payload.longId }.isInstanceOf(NumberFormatException::class.java)
  }
}
